### PR TITLE
Add loss for BridgeTowerForMaskedLM and BridgeTowerForImageAndTextRetrieval

### PR DESCRIPTION
**Pull request by abhiwand**
# What does this PR do?
This PR adds losses to BridgeTowerMaskedLM and BridgeTowerForImageAndTextRetrieval

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sgugger** [comment]
cc @amyeroberts and @younesbelkada

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this! The additions all look good to me. 

Could you add some tests to make sure the loss calculation is working as expected? Once they're included I think it's good to merge 👍

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for this nice addition! 
Yes let's add some tests - I imagine you can implement a very simple test such as this one: https://github.com/huggingface/transformers/blob/main/tests/models/blip/test_modeling_blip.py#LL642C29-L642C29 just to make sure forward/backward works properly

File: src/transformers/models/bridgetower/modeling_bridgetower.py
Comment (regisss) [inline src/transformers/models/bridgetower/modeling_bridgetower.py:1688]: Here we should instantiate the loss function `loss_fct = CrossEntropyLoss()` as for the MLM loss.

Comment (tileintel) [reply src/transformers/models/bridgetower/modeling_bridgetower.py]: Thank @regisss for the suggestion. We have addressed this in the latest commit.

**Comment by tileintel** [comment]
Thank @amyeroberts, @younesbelkada, @regisss for your review and your suggestions. We have addressed your comments and have added few tests for loss computation and for forward/backward as suggested ones. 
Can you please help to merge this PR if possible? Thanks a lot

**Review by younesbelkada — COMMENTED** [review]
Thanks so much for addressing the comments! 
It seems that some quality tests are failing on files that are independent from this PR, could you please rebase or merge your current branch with the `main` branch?

Usually what I do in these situation is (assuming that your upstream remote is `https://github.com/huggingface/transformers.git` - you can check that with `git remote -v`, if you don't see it run `git remote add upstream https://github.com/huggingface/transformers.git`):
```
git fetch upstream
git merge upstream/main
git push
```
If you are more familiar with `rebase` you can also do so

I can also do it for you, let me know

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this and the tests! 

As @younesbelkada mentioned, you'll need to add the updates from main to resolve the `check_code_quality` failing tests. Otherwise, I think it's gtg :)

**Comment by tileintel** [comment]
Thank @amyeroberts for approving this PR.
Thank @younesbelkada for your suggestion. I have resolved the failed quality tests. Can you please approve and merge this PR? c
Thanks a lot

**Review by younesbelkada — APPROVED** [review]
Thanks for the fix! 

I have a slight doubt on the [usage of `AlmostEqual`](https://github.com/abhiwand/transformers/blob/95dc61ce9c82c1a44cacfad8179ee73724c854c3/tests/models/bridgetower/test_modeling_bridgetower.py#L428) in favor of `torch.all_close`, but I don't have any strong opinion, I think this is perfectly fine. Let's see what @ydshieh & @sgugger will say

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

**Pull request merged**
